### ISSUE TITLE
Override return_name when creating a shipment

### ISSUE THREAD

**Issue by joshprintsimple** [chitchats/chitchats-api-doc#63]
Hi, I've been writing an integration to ChitChats and I'd like to be able to specify the 'return_name' when creating a shipment. Is there a way to do this, or any of the return_address fields for that matter? Or does it automatically fill all of the values for the 'return_' fields from the user's account?

I get this error if I only specify the return_name:
"return_province_code field is invalid, return_postal_code field can't be blank, return_address_1 field can't be blank, and return_city field can't be blank"

If I provide the entire return address along with the return_name, using a valid address (which I can provide if necessary) I get this error:
"return_province_code field is invalid and return_postal_code field is invalid"
I am using a two letter province code and the postal code I'm using does have the space in this scenario in case that matters.

Can anyone help with this?

**Comment by mlywe** [comment]
https://github.com/chitchats/chitchats-api-doc/blob/master/sections/shipments.md#create-a-shipment
Only U.S. return address is supported when setting it through API.

Here is a [sample](https://gist.github.com/mlywe/4377528fa7e5d8af67129fc14ab474ea) payload using curl. Feel free to try it against our staging env e.g. staging.chitchats.com (you have to sign up first if you don't have account there).

**Comment by DavePrintSimple** [comment]
@mlywe 

Thank you for your response.

We integrated with Chit Chats to help our clients, who run printing businesses and fulfill orders on platforms like Etsy, Shopify, and Amazon. This integration allows them to use their Chit Chats account directly on our platform, eliminating the need for the Chit Chats dashboard. Our software assists them in managing their orders, and generating Chit Chats shipping labels from our platform helps streamline their workflow.

Our clients often dropship or print orders for customers selling on Shopify, Etsy, or Amazon. To ensure the end customer receives the package as if it's directly from the online store, our clients need the flexibility to print their client's name on the shipping label. To achieve this, we found that we have to fill out return information to overwrite names on a shipment-by-shipment basis.

However, based on your response, it seems we also need to specify a US return address regardless of the shipment's destination. For example, if the package is sent via Chit Chats Select service, we add a US address, and the package is directed to your US returns facility if undeliverable?

Please let us know if we're understanding that correctly.

Thanks,
Dave

**Comment by mlywe** [comment]
> However, based on your response, it seems we also need to specify a US return address regardless of the shipment's destination. For example, if the package is sent via Chit Chats Select service, we add a US address, and the package is directed to your US returns facility if undeliverable?

No, you don't set `return_us_address` or any of the return field for CA shipment e.g. Chit Chats Select. Currently, we only support custom U.S. return address for U.S. shipment, and not for CA shipment.

For undeliverable CA shipment and if Client wishes to get their shipment returned to them:
- Clients would have to sign in to their Chit Chats account and set their return method on https://chitchats.com/clients/_/settings/returns
- currently we only support branch pickup so Client would have to visit their selected branch to pick up the return shipment
  - and there is processing fee for return shipment

**Comment by DavePrintSimple** [comment]
@mlywe 
Thanks for the information. However, this still doesn't answer our question.

How can we change the 'Ship From' name on the shipping label without having to specify the return address? Or is that not possible and you can only print the 'Ship From' name on the Chit Chats account the shipping label was created from?

**Comment by mlywe** [comment]
> How can we change the 'Ship From' name on the shipping label without having to specify the return address? Or is that not possible and you can only print the 'Ship From' name on the Chit Chats account the shipping label was created from?

Yes, that is correct. Not possible to do so for CA shipment as the `Ship From` name is taken from the account's business name (or profile name) which you cannot modify through API for your usecase.
